rdkit/Chem/SaltRemover.py

```diff
@@ -168,7 +170,7 @@ def _applyPattern(m, salt, notEverything):
     if dontRemoveEverything and len(Chem.GetMolFrags(mol)) <= 1:
       return mol
     modified = False
-    for i, salt in enumerate(self.salts):
+    for salt in self.salts:
       tMol = _applyPattern(mol, salt, dontRemoveEverything)
       if tMol is not mol:
```

Comment:
I think that the check is supposed to see if the molecule has changed when applying the pattern. I think this can happen when the salt pattern is not found in mol.